### PR TITLE
Extract StoreType from NeoStores.

### PR DESCRIPTION
**Pull request by MishaDemianenko**
Finish code cleanup on neostores after removal of possibility to create stores lazilly.
Introduce separate StoreType enum, provide way of mapping between store file name and StoreType.

### CODE REVIEW

**Comment by MishaDemianenko** [comment]
@chrisvest here is the extraction of storeType from a NeoStores, leftover from lazy store creation removal story. Can you please take a look?

File: community/kernel/src/main/java/org/neo4j/kernel/impl/store/StoreType.java [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+import java.io.IOException;
+
+import org.neo4j.graphdb.factory.GraphDatabaseSettings;
+import org.neo4j.kernel.IdType;
+import org.neo4j.kernel.impl.store.counts.CountsTracker;
+
+public enum StoreType
+{
+    NODE_LABEL( StoreFactory.NODE_LABELS_STORE_NAME )
+            {
```
Comment (tinwelint) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/store/StoreType.java:31]: I think we should fix this code style in IntelliJ where it indents 2 "tabs" extract for everything for enums. It's quite unnecessary.

Comment (thobe) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/store/StoreType.java:31]: I've tried fixing that for years, I even reported an issue about it to the Intellij guys, to no use. With the opening curly on a new line Intellij (incorrectly) treats this as a "continuation line" and indents as such.

Comment (MishaDemianenko) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/store/StoreType.java:31]: @tinwelint just to confirm - it's still not possible. There are couple of open issues/requests regarding enum formatting in bug tracker (ex. https://youtrack.jetbrains.com/issue/IDEA-106353), but no progress so far.

File: community/kernel/src/main/java/org/neo4j/kernel/impl/store/NeoStores.java [hunk 2]
```diff
@@ -694,4 +440,132 @@ public void assertOpen()
             throw new IllegalStateException( "Database has been shutdown" );
         }
     }
+
+    CommonAbstractStore createDynamicStore(String storeName, IdType idType, Setting<Integer> blockSizeProperty)
```
Comment (chrisvest) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/store/NeoStores.java]: Parameter list spaces

File: community/kernel/src/main/java/org/neo4j/kernel/impl/store/NeoStores.java [hunk 3]
```diff
@@ -694,4 +440,132 @@ public void assertOpen()
...
+    {
+        return createDynamicStore( storeName, idType, config.get( blockSizeProperty ) );
+    }
+
+    CommonAbstractStore createDynamicStore(String storeName, IdType idType, int blockSize)
```
Comment (chrisvest) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/store/NeoStores.java]: Parameter list spaces

File: community/kernel/src/main/java/org/neo4j/kernel/impl/store/NeoStores.java [hunk 4]
```diff
@@ -694,4 +440,132 @@ public void assertOpen()
...
+    {
+        File storeFile = getStoreFile( storeName );
+        return initialize( new DynamicArrayStore( storeFile, config, idType, idGeneratorFactory, pageCache,
+                logProvider, blockSize ) );
+    }
+
+    CommonAbstractStore createNeoStore( String storeName )
```
Comment (chrisvest) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/store/NeoStores.java]: This should be called `createNodeStore`

File: community/kernel/src/main/java/org/neo4j/kernel/impl/store/NeoStores.java [hunk 5]
```diff
@@ -694,4 +440,132 @@ public void assertOpen()
...
+    {
+        File storeFile = getStoreFile( storeName );
+        return initialize( new NodeStore( storeFile, config, idGeneratorFactory, pageCache, logProvider,
+                (DynamicArrayStore) getOrCreateStore(StoreType.NODE_LABEL) ) );
```
Comment (chrisvest) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/store/NeoStores.java]: Parameter list spaces in getOrCreateStore

File: community/kernel/src/main/java/org/neo4j/kernel/impl/store/NeoStores.java (hunk 2 above)
Comment (chrisvest) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/store/NeoStores.java]: Maybe call these `createDynamicArrayStore` to mirror `createDynamicStringStore`

File: community/kernel/src/main/java/org/neo4j/kernel/impl/store/NeoStores.java [hunk 6]
```diff
@@ -694,4 +440,132 @@ public void assertOpen()
...
+    {
+        File fileName = getStoreFile( storeName );
+        return initialize( new SchemaStore( fileName, config, IdType.SCHEMA, idGeneratorFactory, pageCache,
+                logProvider ) );
+    }
+
+    CommonAbstractStore createMetadataStore()
+    {
+        return initialize( new MetaDataStore( neoStoreFileName, config, idGeneratorFactory, pageCache, logProvider ) );
+    }
```
Comment (chrisvest) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/store/NeoStores.java:569]: I'd move `createMetaDataStore` bellow `createCountsStore` so the order roughly matches the enum.

**Comment by chrisvest** [comment]
@MishaDemianenko Had a number of small nits, but otherwise looks good.

**Comment by MishaDemianenko** [comment]
@chrisvest all comments should be fixed now.

**Pull request merged**